foundationdb/src/database.rs

```diff
@@ -64,37 +65,55 @@ impl Database {
     /// It might retry indefinitely if the transaction is highly contentious. It is recommended to
     /// set `TransactionOption::RetryLimit` or `TransactionOption::SetTimeout` on the transaction
     /// if the task need to be guaranteed to finish.
-    pub fn transact<F, Fut, Item>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = FdbError>>
+    pub fn transact<F, Fut, Item, E>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = E>>
     where
         F: FnMut(Transaction) -> Fut + 'static,
-        Fut: Future<Item = Item, Error = FdbError> + 'static,
+        Fut: Future<Item = Item, Error = E> + 'static,
         Item: 'static,
+        E: From<Error> + Send + Sync + 'static,
     {
+        // This will be used to temporarily capture the internal error of the FnMut, F.
+        enum EitherError<E> {
```

Comment:
@yjh0502 I think this is needed to temporarily capture the raw error during execution.

Reply:
I think we can do it without introducing another error type.

https://github.com/yjh0502/foundationdb-rs/commit/838ecb2387b7eb57ff890489d100daaf38d7925d

Reply:
Ah yes... nice work.